tofu/modules/harvester_host/locals.tf

```diff
@@ -0,0 +1,57 @@
+locals {
+  nonexistent_cloudinit_secrets     = [for cloudinit in var.cloudinit_secrets : cloudinit if length(cloudinit.user_data) > 0]
+  nonexistent_cloudinit_secrets_map = { for cloudinit in local.nonexistent_cloudinit_secrets : cloudinit.name => cloudinit }
+  existing_cloudinit_secrets        = [for cloudinit in var.cloudinit_secrets : cloudinit if !contains(local.nonexistent_cloudinit_secrets, cloudinit)]
+  existing_cloudinit_secrets_map    = { for cloudinit in local.existing_cloudinit_secrets : cloudinit.name => cloudinit }
+  default_init                      = <<EOT
+#cloud-config
```

Comment:
How about using an external template file, which could be easier to understand? See aws_host/user_data.yaml